api/swig/python/CMakeLists.txt

```diff
@@ -0,0 +1,50 @@
+set(CMAKE_CXX_STANDARD 14)
+find_package(Python3 COMPONENTS Interpreter Development REQUIRED)
+include_directories(../..)
+
+set(CMAKE_SWIG_FLAGS "-Wextra" "-v" "-py3")
+
+set_source_files_properties("bagpy.i" PROPERTIES
+    SWIG_MODULE_NAME bagPy
+    CPLUSPLUS ON
+    COMPILE_FLAGS "--std=c++14"
+)
+
+# Proper Python exception generation results in larger code, which MSVC can't handle by default 
+if (MSVC)
+    add_compile_options(/bigobj)
+endif()
+
+swig_add_library(bagPy TYPE SHARED LANGUAGE python
+    SOURCES "bagpy.i"
+)
+
+swig_link_libraries(bagPy
+    baglib
+    Python3::Python
+)
+
+if(${CMAKE_SYSTEM_NAME} MATCHES Darwin)
+    add_custom_command(TARGET bagPy POST_BUILD
+            COMMAND ${CMAKE_COMMAND} -E copy_if_different $<TARGET_FILE:bagPy> "_bagPyd.so")
+    set(BAGPY_LIB _bagPyd.so)
+elseif(${CMAKE_SYSTEM_NAME} MATCHES Linux)
+    set(BAGPY_LIB _bagPy.so)
+elseif(${CMAKE_SYSTEM_NAME} MATCHES Windows)
+    set(BAGPY_LIB _bagPy.pyd)
+endif()
+
+# Files to install with Python
+set(PYTHON_INSTALL_FILES
+        ${CMAKE_CURRENT_BINARY_DIR}/bagPy.py
+        ${CMAKE_CURRENT_BINARY_DIR}/${BAGPY_LIB})
+
+# Configure setup.py and copy to output directory
+set(SETUP_PY_IN ${CMAKE_CURRENT_SOURCE_DIR}/setup.py.in)
+set(SETUP_PY_OUT ${CMAKE_CURRENT_BINARY_DIR}/setup.py)
+configure_file(${SETUP_PY_IN} ${SETUP_PY_OUT})
+
+# Install target to call setup.py
+add_custom_target(install-python
+        DEPENDS _bagPy
+        COMMAND python ${SETUP_PY_OUT} install)
```

Comment:
This custom target is not working on Windows (and BAG_CI being OFF) because of the _bagPy.pyd location:

```
1>------ Build started: Project: install-python, Configuration: Release x64 ------
1>running install
1>C:\Users\gmasetti\AppData\Local\HyOc\py3\hyo2\envs\pydro38\lib\site-packages\setuptools\dist.py:493: UserWarning: Normalizing '2.0.1-dev0' to '2.0.1.dev0'
1>  warnings.warn(tmpl.format(**locals()))
1>C:\Users\gmasetti\AppData\Local\HyOc\py3\hyo2\envs\pydro38\lib\site-packages\setuptools\command\install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
1>  warnings.warn(
1>CUSTOMBUILD : error : [Errno 2] No such file or directory: 'C:/code/cpp/v2_build/api/swig/python/_bagPy.pyd'
1>Done building project "install-python.vcxproj" -- FAILED.
```

Reply:
Will address in 2.0.2. See: https://github.com/OpenNavigationSurface/BAG/issues/69